vtex-sites/base.store
# feat: Implement pending features on Out of Stock

## What's the purpose of this pull request?

This PR was created to implement the pending features of the Out of Stock component. 


**Success transitions:**

https://user-images.githubusercontent.com/10616243/159769503-f6576e70-fb7b-4688-9886-973bbe3a1477.mp4


## How does it work?

The `OutOfStock` component is displayed only in products that are not available. The loading and success states will be displayed in the button, when the user submit his email. At the moment error messages are not being displayed, this will be tackled then the `Input` component is updated (error message will be displayed below the input).  

## How to test it?

One way you can see the component in the page is opening the PDP of a product that is unavailable (which you probably won't find in Base.Store) or you can remove the `!` from the `!availability &&` in line 179 of `ProductDetails.tsx` file (just for testing purposes).

## References

Initial PR for the Out of Stock #146 
Related [Jira Task](https://vtex-dev.atlassian.net/browse/FSSS-95)
Related [Figma file](https://www.figma.com/file/zHMOtOPduiM0hNNcg21iR1/Handoff-v1(PDP)?node-id=110%3A37849)

## Checklist
- [x] Style Out of Stock
- [x] Add icon to button
- [x] Add success transitions
